encoder.go

```diff
@@ -55,12 +49,15 @@ func encodeTriggerBody(
 	if eventExceedsMaximumSize {
 		return nil, errors.New(fmt.Sprintf("Event payload exceeded maximum size (%d bytes is too much)", len(payloadData)))
 	}
-	return json.Marshal(&eventPayload{
-		Name:     event,
-		Channels: channels,
-		Data:     payloadData,
-		SocketID: socketID,
-	})
+	eventPayload := map[string]interface{}{
+		"name":     event,
+		"channels": channels,
+		"data":     payloadData,
+	}
+	for k, v := range params {
+		eventPayload[k] = v
+	}
```

Comment:
Just highlighting that, by doing it this way, if a caller accidentally includes a `params` `map` with a `name`/`channels`/`data` key it will override the named args

Reply:
Good point. I hadn't thought of that. I'll add a check because even though I think this is reasonable behaviour, there's a very small chance because it could lead to some very confusing bugs.